### PR TITLE
chore!: Initial gitlab setup

### PR DESCRIPTION
**Pull request by MetCoder95**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  extends:
+    - .base-stage
+  image: eu.gcr.io/taxfix-central/devops-tools:latest
+  script:
+    - docker push $REGISTRY_IMAGE
+  rules:
+    - if: $CI_COMMIT_BRANCH =~ /^preview/
+      when: on_success
+  environment:
+    name: preview
```
Comment (ngoduykhanh) [inline .gitlab-ci.yml:90]: As far as I understand, this pipeline setup is just for building and pushing the docker image.
Do we need the `deploy-preview`, `deploy-integration`, `deploy-staging` and `deploy-production` here?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      IMAGE_TAG=$RELEASE_VERSION-sha.${CI_COMMIT_SHA}
+    fi
+  - export IMAGE_TAG=$IMAGE_TAG
+  - |
+    if ([[ "$CI_COMMIT_BRANCH" =~ ^(master|main)$ ]] && [[ ! -z $CI_COMMIT_TAG ]])
+    then
+      export REGISTRY_IMAGE=${REGISTRY_URL}/$PROJECT_NAME/${CI_PROJECT_NAME}:$IMAGE_TAG
+    else
+      export REGISTRY_IMAGE=${REGISTRY_URL}/$PROJECT_NAME/${CI_PROJECT_NAME}:$IMAGE_TAG-$BUILD_ENV
+      export REGISTRY_IMAGE_LATEST=${REGISTRY_URL}/$PROJECT_NAME/${CI_PROJECT_NAME}:latest
```
Comment (ngoduykhanh) [inline .gitlab-ci.yml:49]: I think this script needs to be adjusted as well. Let's remove unnecessary code, e.g mono-repo, preview?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,147 @@
+variables:
+  REGISTRY_URL: "eu.gcr.io"
+  DOCKER_HOST: "tcp://docker:2375"
```
Comment (ngoduykhanh) [inline .gitlab-ci.yml:3]: Set `DOCKER_HOST: "tcp://dind:2375"` will help the pipeline to use the Docker layer caching.

**Review by ngoduykhanh — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (MetCoder95) [reply .gitlab-ci.yml:90]: You're right, but my idea was to have the chance to properly test it out in the same way as a single service. So if you want to e.g.  **test your docker image and see how it behaves on the cluster**, might be good to have a chance to do so.
Maybe you're right, two steps can be removed like `integration` and `staging`. But `preview` can be nice to have so we can directly test it out. What do you think?

File: .gitlab-ci.yml (hunk 2 above)
Comment (MetCoder95) [reply .gitlab-ci.yml:49]: You're right, didn't do cleanup 🙏

File: .gitlab-ci.yml (hunk 3 above)
Comment (MetCoder95) [reply .gitlab-ci.yml:3]: I knew I was missing something for caching 😅

File: .gitlab-ci.yml (hunk 1 above)
Comment (MetCoder95) [reply .gitlab-ci.yml:86]: I removed `integration` and `staging`, hope is ok 😄

**Comment by MetCoder95** [comment]
I cannot ask to for another review but can you take a second look? @ngoduykhanh 🙏

**Comment by MetCoder95** [comment]
Merging now, if you have more feedback please feel free to make it, I'll apply them when creating `hades` repo 👍

**Pull request merged**